File: thrift-php/TFilterProtocol.php
<?php
/*
 * Copyright 2013 Ke Meng (mengke@icloud.com)
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * Created by JetBrains PhpStorm.
 * User: ibntab
 * Date: 13-7-8
 * Time: 下午3:32
 */
namespace Thrift\Protocol;

use Thrift\Protocol\TProtocol;
use Thrift\Factory\TStringFuncFactory;

//
abstract class TFilterProtocol extends TProtocol {

    private $delegate;

    public function __construct($delegate) {
        parent::__construct($delegate->getTransport());
        $this->delegate = $delegate;
    }
    /**
     * Writes the message header
     *
     * @param string $name Function name
     * @param int $type message type TMessageType::CALL or TMessageType::REPLY
     * @param int $seqid The sequence id of this message
     */
    public function writeMessageBegin($name, $type, $seqid)
    {
        $this->delegate->writeMessageBegin($name, $type, $seqid);
    }

    /**
     * Close the message
     */
    public function writeMessageEnd()
    {
        $this->delegate->writeMessageEnd();
    }

    /**
     * Writes a struct header.
     *
     * @param string $name Struct name
     * @throws TException on write error
     * @return int How many bytes written
     */
    public function writeStructBegin($name)
    {
        $this->delegate->writeStructBegin($name);
    }

    /**
     * Close a struct.
     *
     * @throws TException on write error
     * @return int How many bytes written
     */
    public function writeStructEnd()
    {
        $this->delegate->writeStructEnd();
    }

    public function writeFieldBegin($fieldName, $fieldType, $fieldId)
    {
        $this->delegate->writeFieldBegin($fieldName, $fieldType, $fieldId);
    }

    public function writeFieldEnd()
    {
        $this->delegate->writeFieldEnd();
    }

    public function writeFieldStop()
    {
        $this->delegate->writeFieldStop();
    }

    public function writeMapBegin($keyType, $valType, $size)
    {
        $this->delegate->writeMapBegin($keyType, $valType, $size);
    }

    public function writeMapEnd()
    {
        $this->delegate->writeMapEnd();
    }

    public function writeListBegin($elemType, $size)
    {
        $this->delegate->writeListBegin($elemType, $size);
    }

    public function writeListEnd()
    {
        $this->delegate->writeListEnd();
    }

    public function writeSetBegin($elemType, $size)
    {
        $this->delegate->writeSetBegin($elemType, $size);
    }

    public function writeSetEnd()
    {
        $this->delegate->writeSetEnd();
    }

    public function writeBool($bool)
    {
        $this->delegate->writeBool($bool);
    }

    public function writeByte($byte)
    {
        $this->delegate->writeByte($byte);
    }

    public function writeI16($i16)
    {
        $this->delegate->writeI16($i16);
    }

    public function writeI32($i32)
    {
        $this->delegate->writeI32($i32);
    }

    public function writeI64($i64)
    {
        $this->delegate->writeI64($i64);
    }

    public function writeDouble($dub)
    {
        $this->delegate->writeDouble($dub);
    }

    public function writeString($str)
    {
        $this->delegate->writeString($str);
    }

    /**
     * Reads the message header
     *
     * @param string $name Function name
     * @param int $type message type TMessageType::CALL or TMessageType::REPLY
     * @parem int $seqid The sequence id of this message
     */
    public function readMessageBegin(&$name, &$type, &$seqid)
    {
        $this->delegate->readMessageBegin($name, $type, $seqid);
    }

    /**
     * Read the close of message
     */
    public function readMessageEnd()
    {
        $this->delegate->readMessageEnd();
    }

    public function readStructBegin(&$name)
    {
        $this->delegate->readStructBegin($name);
    }

    public function readStructEnd()
    {
        $this->delegate->readStructEnd();
    }

    public function readFieldBegin(&$name, &$fieldType, &$fieldId)
    {
        $this->delegate->readFieldBegin($name, $fieldType, $fieldId);
    }

    public function readFieldEnd()
    {
        $this->delegate->readFieldEnd();
    }

    public function readMapBegin(&$keyType, &$valType, &$size)
    {
        $this->delegate->readMapBegin($keyType, $valType, $size);
    }

    public function readMapEnd()
    {
        $this->delegate->readMapEnd();
    }

    public function readListBegin(&$elemType, &$size)
    {
        $this->delegate->readListBegin($elemType, $size);
    }

    public function readListEnd()
    {
        $this->delegate->readListEnd();
    }

    public function readSetBegin(&$elemType, &$size)
    {
        $this->delegate->readSetBegin($elemType, $size);
    }

    public function readSetEnd()
    {
        $this->delegate->readSetEnd();
    }

    public function readBool(&$bool)
    {
        $this->delegate->readBool($bool);
    }

    public function readByte(&$byte)
    {
        $this->delegate->readByte($byte);
    }

    public function readI16(&$i16)
    {
        $this->delegate->readI16($i16);
    }

    public function readI32(&$i32)
    {
        $this->delegate->readI32($i32);
    }

    public function readI64(&$i64)
    {
        $this->delegate->readI64($i64);
    }

    public function readDouble(&$dub)
    {
        $this->delegate->readDouble($dub);
    }

    public function readString(&$str)
    {
        $this->delegate->readString($str);
    }
}

File: las-server/src/main/java/org/apache/thrift/protocol/TFilterProtocol.java
/*
 * Copyright 2013 Ke Meng (mengke@icloud.com)
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 */

package org.apache.thrift.protocol;

import org.apache.thrift.TException;

import java.nio.ByteBuffer;

/**
 * Created with IntelliJ IDEA.
 * User: ibntab
 * Date: 13-7-8
 * Time: 下午4:27
 */
public abstract class TFilterProtocol extends TProtocol {

    private final TProtocol delegate;

    public TFilterProtocol(TProtocol tProtocol) {
        super(tProtocol.getTransport());
        this.delegate = tProtocol;
    }

    @Override
    public void writeMessageBegin(TMessage tMessage) throws TException {
        delegate.writeMessageBegin(tMessage);
    }

    @Override
    public void writeMessageEnd() throws TException {
        delegate.writeMessageEnd();
    }

    @Override
    public void writeStructBegin(TStruct tStruct) throws TException {
        delegate.writeStructBegin(tStruct);
    }

    @Override
    public void writeStructEnd() throws TException {
        delegate.writeStructEnd();
    }

    @Override
    public void writeFieldBegin(TField tField) throws TException {
        delegate.writeFieldBegin(tField);
    }

    @Override
    public void writeFieldEnd() throws TException {
        delegate.writeFieldEnd();
    }

    @Override
    public void writeFieldStop() throws TException {
        delegate.writeFieldStop();
    }

    @Override
    public void writeMapBegin(TMap tMap) throws TException {
        delegate.writeMapBegin(tMap);
    }

    @Override
    public void writeMapEnd() throws TException {
        delegate.writeMapEnd();
    }

    @Override
    public void writeListBegin(TList tList) throws TException {
        delegate.writeListBegin(tList);
    }

    @Override
    public void writeListEnd() throws TException {
        delegate.writeListEnd();
    }

    @Override
    public void writeSetBegin(TSet tSet) throws TException {
        delegate.writeSetBegin(tSet);
    }

    @Override
    public void writeSetEnd() throws TException {
        delegate.writeSetEnd();
    }

    @Override
    public void writeBool(boolean b) throws TException {
        delegate.writeBool(b);
    }

    @Override
    public void writeByte(byte b) throws TException {
        delegate.writeByte(b);
    }

    @Override
    public void writeI16(short i) throws TException {
        delegate.writeI16(i);
    }

    @Override
    public void writeI32(int i) throws TException {
        delegate.writeI32(i);
    }

    @Override
    public void writeI64(long l) throws TException {
        delegate.writeI64(l);
    }

    @Override
    public void writeDouble(double v) throws TException {
        delegate.writeDouble(v);
    }

    @Override
    public void writeString(String s) throws TException {
        delegate.writeString(s);
    }

    @Override
    public void writeBinary(ByteBuffer byteBuffer) throws TException {
        delegate.writeBinary(byteBuffer);
    }

    @Override
    public TMessage readMessageBegin() throws TException {
        return delegate.readMessageBegin();
    }

    @Override
    public void readMessageEnd() throws TException {
        delegate.readMessageEnd();
    }

    @Override
    public TStruct readStructBegin() throws TException {
        return delegate.readStructBegin();
    }

    @Override
    public void readStructEnd() throws TException {
        delegate.readStructEnd();
    }

    @Override
    public TField readFieldBegin() throws TException {
        return delegate.readFieldBegin();
    }

    @Override
    public void readFieldEnd() throws TException {
        delegate.readFieldEnd();
    }

    @Override
    public TMap readMapBegin() throws TException {
        return delegate.readMapBegin();
    }

    @Override
    public void readMapEnd() throws TException {
        delegate.readMapEnd();
    }

    @Override
    public TList readListBegin() throws TException {
        return delegate.readListBegin();
    }

    @Override
    public void readListEnd() throws TException {
        delegate.readListEnd();
    }

    @Override
    public TSet readSetBegin() throws TException {
        return delegate.readSetBegin();
    }

    @Override
    public void readSetEnd() throws TException {
        delegate.readSetEnd();
    }

    @Override
    public boolean readBool() throws TException {
        return delegate.readBool();
    }

    @Override
    public byte readByte() throws TException {
        return delegate.readByte();
    }

    @Override
    public short readI16() throws TException {
        return delegate.readI16();
    }

    @Override
    public int readI32() throws TException {
        return delegate.readI32();
    }

    @Override
    public long readI64() throws TException {
        return delegate.readI64();
    }

    @Override
    public double readDouble() throws TException {
        return delegate.readDouble();
    }

    @Override
    public String readString() throws TException {
        return delegate.readString();
    }

    @Override
    public ByteBuffer readBinary() throws TException {
        return delegate.readBinary();
    }
}
